client/my-sites/site-settings/settings-security/controller.js

```diff
@@ -1,7 +1,17 @@
+import page from 'page';
 import { createElement } from 'react';
 import SecurityMain from 'calypso/my-sites/site-settings/settings-security/main';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
 
 export function security( context, next ) {
-	context.primary = createElement( SecurityMain );
-	next();
+	const state = context.store.getState();
+	const siteId = getSelectedSiteId( state );
+	// If we have a site ID, render the main component
+	// Otherwise, redirect to site selection.
+	if ( siteId ) {
+		context.primary = createElement( SecurityMain );
+		next();
```

Comment:
It's also a good idea to return this too:

```suggestion
		return next();
```

which will be necessary if anyone adds anything below the `if/else` statement and we don't want to inadvertently execute it.

Reply:
Got it, updated!